### PR TITLE
Make topBlockHeight and balance observables

### LINKED ISSUE #646: Issue with claiming
While claiming tip was successful, the balance of my wallet has not changed (last transaction 11/17/20):

Безіменний 0004

I noticed that the wallet makes a transaction with a tip claim, increasing the amount. I sent 0.1 tokens, 0.2 is claimed (0.1 was already). Then I sent another 0.1 tokens, claimed already 0.3 tokens. It's strange. The balance in my wallet has not changed.

Superhero d311677 / v.0.4.2, 
Chrome 87.0.4280.67

### PR DESCRIPTION
**Pull request by CedrikNikita**
fixes #646, #641
potential fix for #629

### CODE REVIEW

File: src/popup/router/components/AmountSend.vue [hunk 1]
```diff
@@ -23,7 +23,7 @@
       <div class="balance-box">
         <p class="label">{{ $t('pages.tipPage.availableLabel') }}</p>
         <span class="secondary-text f-14 block l-1" data-cy="balance">
-          {{ selectedToken ? selectedToken.convertedBalance : tokenBalance }}
+          {{ selectedToken ? selectedToken.convertedBalance : tokenBalance.toFixed(2) }}
```
Comment (mradkov) [inline src/popup/router/components/AmountSend.vue:26]: Why `.toFixed(2)`?

The fungible tokens can have as many decimal pieces as they want to. 🤔

File: src/popup/router/components/BalanceInfo.vue [hunk 2]
```diff
@@ -11,7 +11,7 @@
             is-custom
           />
           <span class="display-value text-ellipsis">{{
-            selectedToken ? selectedToken.convertedBalance : tokenBalance
+            selectedToken ? selectedToken.convertedBalance : tokenBalance.toFixed(2)
```
Comment (mradkov) [inline src/popup/router/components/BalanceInfo.vue:14]: Same - why `toFixed(2)` 👆?

File: src/popup/router/components/AmountSend.vue (hunk 1 above)
Comment (CedrikNikita) [reply src/popup/router/components/AmountSend.vue:26]: https://github.com/aeternity/superhero-wallet/pull/652/files#diff-42d961a783c297603d8cd3d38fb7471e62b81eb03a5ac50e7cb3124042e4b495L44-L48

Comment (mradkov) [reply src/popup/router/components/AmountSend.vue:26]: Ok. Then we need to discuss this functionality, but not in the scope of this PR.

File: src/store/plugins/observables.js [hunk 3]
```diff
@@ -19,6 +20,31 @@ export default (store) => {
...
+            })
+            .then((result) => {
+              const balance = convertToAE(result);
+              if (balance !== store.state.balance) {
+                store.commit('updateBalance', balance);
+              }
+              return balance;
+            })
+        : Promise.resolve(store.state.balance),
+    ),
```
Comment (davidyuk) [inline src/store/plugins/observables.js:43]: We can write it simpler filtering out the `null` values of sdk:
```suggestion
    switchMap((sdk) =>
      sdk.balance(store.state.account.publicKey).catch((error) => {
        if (!isNotFoundError(error)) {
          handleUnknownError(error);
        }
        return 0;
      }),
    ),
    map((balanceAettos) => {
      const balance = convertToAE(balanceAettos);
      if (balance !== store.state.balance) {
        store.commit('updateBalance', balance);
      }
      return balance;
    }),
```
So, proposing to rewrite `sdk$` from the above as:
```js
  const sdk$ = watchAsObservable(({ sdk }) => sdk, { immediate: true }).pipe(
    pluck('newValue'),
    filter((sdk) => sdk),
  );
```

File: src/store/plugins/observables.js [hunk 4]
```diff
@@ -19,6 +20,31 @@ export default (store) => {
...
+    multicast(new BehaviorSubject(store.state.balance)),
```
Comment (davidyuk) [inline src/store/plugins/observables.js:44]: The arguments to don't store the balance in store:
- developers may use the balance from the store not aware that it won't be updated without a subscription to the corresponding observable variable
- balance polling emits wasteful mutations that can complicate debugging
- the cached balance not needed to be handled the same careful (don't need to maintain in migrations) as other state fields persisted in local storage

Proposing to sync it with local storage explicitly, using a different key. For example, to get the balance by:
```js
JSON.parse(localStorage['rxjs']).balance
```
and to set it by:
```js
localStorage['rxjs'] = JSON.stringify({ ...JSON.parse(localStorage['rxjs']), balance })
```
even better would be to write a helper functions for that

File: src/store/plugins/observables.js [hunk 5]
```diff
@@ -41,5 +67,28 @@ export default (store) => {
...
 
   // eslint-disable-next-line no-param-reassign
-  store.state.observables = { notifications: notifications$ };
+  store.state.observables = {
+    notifications: notifications$,
+    balance: balance$,
+    topBlockHeight: createSdkObservable(async (sdk) => (await sdk.topBlock()).height, 0),
+    tokenBalance: watchAsObservable(
+      (state) => ({
+        token: state.current.token,
```
[the remark this answers is not in the document]
Comment (CedrikNikita) [reply src/store/plugins/observables.js]: Yeah, this PR is more than 1 month old, wallet got new changes

**Review by davidyuk — APPROVED** [review]

**Pull request merged**